__tests__/screens/send-bitcoin-screen.tsx

```diff
@@ -0,0 +1,342 @@
+    ).not.toBeNull()
+
+    fireEvent(amountInput, "onUpdateAmount", 25211)
+
+    expect(
+      queryByText(translate("SendBitcoinScreen.totalExceed", { balance: "$46.64" })),
+    ).toBeNull()
+  })
+
+  it("does not send payment when total exceeds the balance", async () => {
+    const { getByPlaceholderText, getByText, queryByText } = render(
+      <MockedProvider mocks={payKeysendUsernameMocks} cache={cache}>
+        <SendBitcoinScreen route={{ params: null }} />
+      </MockedProvider>,
+    )
+
+    const amountInput = getByText("0.00")
+    const destinationInput = getByPlaceholderText(translate("SendBitcoinScreen.input"))
+
+    fireEvent.changeText(destinationInput, "Bitcoin")
+    fireEvent(amountInput, "onUpdateAmount", 252119)
+
+    const sendButton = getByText(translate("common.send"))
+    fireEvent.press(sendButton)
+
+    await act(async () => await new Promise((resolve) => setTimeout(resolve, 0)))
+    expect(queryByText(translate("SendBitcoinScreen.success"))).toBeNull()
+  })
+
+  it("successfully sends payment by payKeysendUsername", async () => {
+    const { getByPlaceholderText, getByText, queryByText } = render(
+      <MockedProvider mocks={payKeysendUsernameMocks} cache={cache}>
+        <SendBitcoinScreen route={{ params: null }} />
+      </MockedProvider>,
+    )
+
+    const amountInput = getByText("0.00")
+    const destinationInput = getByPlaceholderText(translate("SendBitcoinScreen.input"))
+
+    fireEvent.changeText(destinationInput, "Bitcoin")
+    fireEvent(amountInput, "onUpdateAmount", 25211)
+
+    const sendButton = getByText(translate("common.send"))
+    fireEvent.press(sendButton)
+
+    await act(async () => await new Promise((resolve) => setTimeout(resolve, 0)))
+    expect(queryByText(translate("SendBitcoinScreen.success"))).not.toBeNull()
+  })
+
+  it("successfully sends lightning payment with `lightning:` prefix", async () => {
+    cache.writeQuery({
+      query: WALLET,
+      data: {
+        wallet: [
+          {
+            __typename: "Wallet",
+            balance: 1175855,
+            currency: "BTC",
+            id: "BTC",
+            transactions: transactions,
+          },
+        ],
+      },
+    })
+
+    const lightningPayMocks = [
+      {
+        request: {
+          query: LIGHTNING_PAY,
+          variables: {
+            invoice:
+              "lnbc6864270n1p05zvjjpp5fpehvlv3dd2r76065r9v0l3n8qv9mfwu9ryhvpj5xsz3p4hy734qdzhxysv89eqyvmzqsnfw3pxcmmrddpx7mmdypp8yatwvd5zqmmwypqh2em4wd6zqvesyq5yyun4de3ksgz0dek8j2gcqzpgxqrrss6lqa5jllvuglw5tpsug4s2tmt5c8fnerr95fuh8htcsyx52cp3wzswj32xj5gewyfn7mg293v6jla9cz8zndhwdhcnnkul2qkf6pjlspj2nl3j",
+          },
+        },
+        result: {
+          data: {
+            invoice: {
+              payInvoice: "success",
+              __typename: "Invoice",
+            },
+          },
+        },
+      },
+    ]
+
+    moment.now = function () {
+      return 1598110996000 // Aug 22 2020 10:43
+    }
```

Comment:
Not very familiar with `moment` (which I think we should git rid of, really), but is this time traveling for something that uses `moment` in the code? Can we maybe wrap that with a clear function and mock the function instead?

Reply:
This gets a little tricky due to only being able to mock [the exported function and not internal functions](https://stackoverflow.com/a/55193363). I am going to go with importing the module into itself for now, but I am not sure if that is a good practice.